libs/methodologies/bold/rule-processors/credits/src/rewards-distribution/rewards-distribution.helpers.ts

```diff
@@ -0,0 +1,182 @@
+}): void => {
+  const { actors, remainder } = options;
+
+  for (const [key, actor] of actors) {
+    if (actor.actorType === RewardsDistributionActorType.NETWORK) {
+      actors.set(key, {
+        ...actor,
+        amount: remainder.amount.plus(actor.amount).toString(),
+        percentage: remainder.percentage.plus(actor.percentage).toString(),
+      });
+
+      break;
+    }
+  }
+};
+
+const calculateCertificateTotalValue = (certificateValues: BigNumber[]) =>
+  sumBigNumbers(certificateValues);
+
+export const aggregateMassIdCertificatesRewards = (
+  creditsDocumentUnitPrice: NonZeroPositive,
+  resultContentsWithMassIdCertificateValue: ResultContentsWithMassIdCertificateValue[],
+): AggregateMassIdRewards => {
+  const actors: ActorsByActorType = new Map();
+
+  const certificateTotalValue = calculateCertificateTotalValue(
+    resultContentsWithMassIdCertificateValue.map(
+      ({ massIdCertificateValue }) => massIdCertificateValue,
+    ),
+  );
+
+  const creditTotal = formatDecimalPlaces(
+    certificateTotalValue.multipliedBy(creditsDocumentUnitPrice),
+  );
+
+  for (const {
+    massIdCertificateValue,
+    resultContent: { massIdRewards },
+  } of resultContentsWithMassIdCertificateValue) {
+    for (const massIdReward of massIdRewards) {
+      const { actorType, massIdPercentage, participant } = massIdReward;
+      const actor = actors.get(`${actorType}-${participant.id}`);
+
+      const amount = calculateAmount({
+        creditsDocumentUnitPrice,
+        massIdCertificateValue,
+        massIdPercentage,
+        participantAmount: actor?.amount,
+      });
+
+      actors.set(`${actorType}-${participant.id}`, {
+        actorType,
+        amount,
+        participant,
+        percentage: calculateCreditPercentage(
+          new BigNumber(amount),
+          creditTotal,
+        ),
+      });
+    }
+  }
+
+  return {
+    actors,
+    massIdTotalValue: certificateTotalValue,
+  };
+};
+
+export const calculateRewardsDistribution = (ruleSubject: {
+  creditsDocumentUnitPrice: NonZeroPositive;
+  resultContentsWithMassIdCertificateValue: ResultContentsWithMassIdCertificateValue[];
+}): RewardsDistribution => {
+  const { creditsDocumentUnitPrice } = ruleSubject;
+
+  const resultContentsWithMassIdCertificateValue =
+    ruleSubject.resultContentsWithMassIdCertificateValue.map(
+      ({ massIdCertificateValue, resultContent }) => ({
+        massIdCertificateValue,
+        resultContent,
+      }),
+    );
+
+  const { actors, massIdTotalValue } = aggregateMassIdCertificatesRewards(
+    creditsDocumentUnitPrice,
+    resultContentsWithMassIdCertificateValue,
+  );
+
+  const remainder = calculateRemainder({
+    actors,
+    creditsDocumentUnitPrice,
+    massIdTotalValue,
+  });
+
+  addParticipantRemainder({
+    actors,
+    remainder,
+  });
+
+  return {
+    actors: [...actors.values()],
+    creditsUnitPrice: creditsDocumentUnitPrice,
+    massIdTotalValue: massIdTotalValue.toString(),
```

Comment:
Possibly better if we use certificateTotalValue instead of massIdTotalValue to avoid inconsistency.